workflow-bot-app/src/main/java/com/symphony/bdk/workflow/engine/WorkflowEventHandler.java

```diff
@@ -0,0 +1,86 @@
+package com.symphony.bdk.workflow.engine;
+
+import com.symphony.bdk.workflow.engine.camunda.WorkflowDirectGraphCachingService;
+import com.symphony.bdk.workflow.engine.camunda.WorkflowEventToCamundaEvent;
+import com.symphony.bdk.workflow.engine.camunda.audit.AuditTrailLogger;
+import com.symphony.bdk.workflow.engine.executor.ActivityExecutorContext;
+import com.symphony.bdk.workflow.engine.executor.EventHolder;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.StringUtils;
+import org.camunda.bpm.engine.RuntimeService;
+import org.camunda.bpm.engine.impl.history.event.HistoricVariableUpdateEventEntity;
+import org.camunda.bpm.engine.impl.history.event.HistoryEvent;
+import org.camunda.bpm.engine.impl.history.handler.HistoryEventHandler;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+@Component
+public class WorkflowEventHandler implements HistoryEventHandler {
```

Comment:
i kinda like to make this class as an event publisher, using spring publisher, and let AuditTrailLogger and another class to deal respectively with audit trail log and variables, it is more extensible if one day we have other need on this history event, in addition, it will get rid of the cycle dependency, as they are completely decoupled.

Reply:
I do prefer strategy pattern actually, I will give it a try in next commit.

Reply:
👍 , now it s much better.